cmd/influxd/inspect/auth/auth.go

```diff
@@ -0,0 +1,269 @@
+package auth
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/influxdata/influxdb/v2"
+	"github.com/influxdata/influxdb/v2/authorization"
+	"github.com/influxdata/influxdb/v2/bolt"
+	"github.com/influxdata/influxdb/v2/internal/tabwriter"
+	"github.com/influxdata/influxdb/v2/kit/platform/errors"
+	"github.com/influxdata/influxdb/v2/logger"
+	"github.com/influxdata/influxdb/v2/tenant"
+	"github.com/spf13/cobra"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+func NewAuthCommand() *cobra.Command {
+	base := &cobra.Command{
+		Use:   "auth",
+		Short: "Commands for inspecting on-disk authorizations data",
+		Args:  cobra.NoArgs,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmd.PrintErrf("See '%s -h' for help\n", cmd.CommandPath())
+		},
+	}
+
+	base.AddCommand(NewAuthListCommand())
+	base.AddCommand(NewAuthCreateCommand())
+
+	return base
+}
+
+type authListCommand struct {
+	logger   *zap.Logger
+	boltPath string
+	out      io.Writer
+}
+
+func NewAuthListCommand() *cobra.Command {
+	var authCmd authListCommand
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "List on-disk authorization data",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			config := logger.NewConfig()
+			config.Level = zapcore.InfoLevel
+
+			newLogger, err := config.New(cmd.ErrOrStderr())
+			if err != nil {
+				return err
+			}
+			authCmd.logger = newLogger
+			authCmd.out = cmd.OutOrStdout()
+			return authCmd.run()
+		},
+	}
+
+	defaultPath := filepath.Join(os.Getenv("HOME"), "/.influxdbv2/influxd.bolt")
+
+	cmd.Flags().StringVar(&authCmd.boltPath, "bolt-path", defaultPath, "Path to the TSM data directory.")
+
+	return cmd
+}
+
+// Run executes the run command for authListCommand.
+func (cmd *authListCommand) run() error {
+	ctx := context.Background()
+	store := bolt.NewKVStore(cmd.logger.With(zap.String("system", "bolt-kvstore")), cmd.boltPath)
+	if err := store.Open(ctx); err != nil {
+		return err
+	}
+	defer store.Close()
+	tenantStore := tenant.NewStore(store)
+	tenantService := tenant.NewService(tenantStore)
+	authStore, err := authorization.NewStore(store)
+	if err != nil {
+		return err
+	}
+	auth := authorization.NewService(authStore, tenantService)
+	filter := influxdb.AuthorizationFilter{}
+	auths, _, err := auth.FindAuthorizations(ctx, filter)
+	if err != nil {
+		return err
+	}
+
+	return PrintAuth(ctx, cmd.out, auths, tenantService)
+}
+
+type authCreateCommand struct {
+	logger    *zap.Logger
+	boltPath  string
+	out       io.Writer
+	username  string
+	password  string
+	org       string
+	createOrg bool
+}
+
+func NewAuthCreateCommand() *cobra.Command {
+	var authCmd authCreateCommand
+	cmd := &cobra.Command{
+		Use:   "create",
+		Short: "Create a new user with operator token",
```

Comment:
I worry about having `auth create` being a command to create a new user, as `auth` is used everywhere else to mean a token, not a user.

Could we instead have separate subcommand trees?
1. `auth create` (or even `auth create-operator`) only creates a new token w/ operator permissions. It takes a username but no password
2. `user set-password`/`user create` are new commands to handle recovery of user logins

Reply:
That's probably a good idea - then an existing user can get a token and create a new password as long as the username is remembered.